pkg/queue/manager.go

```diff
@@ -65,9 +69,23 @@ func (m *Manager) getQueue(deploymentName string) *Queue {
 	defer m.mtx.Unlock()
 	q, ok := m.queues[deploymentName]
 	if !ok {
-		q = New(m.concurrencyPerReplica)
+		q = New(int(m.defaultConcurrencyPerReplica), m.defaultConcurrencyPerReplica)
 		m.queues[deploymentName] = q
 		go q.Start()
 	}
 	return q
 }
+
+func (r *Manager) GetCurrencyPerReplica(deployment string) uint32 {
+	return r.getQueue(deployment).concurrencyPerReplica
+}
+
+// SetCurrencyPerReplica updates the concurrency value per replica for a specific deployment.
+// If the provided value is 0, it sets the concurrency value to the default value.
+// This function updates the concurrency value directly on the queue associated with the deployment.
+func (r *Manager) SetCurrencyPerReplica(deploymentName string, newPerReplica uint32) {
```

Comment:
typo? should this be `SetConcurrencyPerReplica` instead of `SetCurrencyPerReplica`?